pubsub/subscriptions/create_bigquery_subscription.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_create_bigquery_subscription]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// createBigQuerySubscription creates a Pub/Sub subscription that exports messages to BigQuery.
+func createBigQuerySubscription(w io.Writer, projectID, subID string, topic *pubsub.Topic, table string) error {
+	// projectID := "my-project-id"
+	// subID := "my-sub"
+	// topic of type https://godoc.org/cloud.google.com/go/pubsub#Topic
+	// table := "my-project-id.dataset_id.table_id"
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	sub, err := client.CreateSubscription(ctx, subID, pubsub.SubscriptionConfig{
+		Topic: topic,
+		BigQueryConfig: pubsub.BigQueryConfig{
+			Table: table,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("client.CreateSubscription: %v", err)
+	}
+	fmt.Fprintf(w, "Created BigQuery subscription: %v\n", sub)
+
+	cfg, err := sub.Config(ctx)
+	if err != nil {
+		return fmt.Errorf("subscription.Config: %v", err)
+	}
+	if cfg.BigQueryConfig.State == pubsub.SubscriptionStateActive {
```

Comment:
We didn't really show the active state in any of the other samples. Also, it is highly improbable that it would ever get out of the active state by the time this returns. To be perfectly honest, I'm not sure that it makes sense to interact with the state programmatically, so I'd just exclude it from the sample.

Reply:
Yeah I thought it'd be nice to show this behavior but it might be too much for the sample. I removed it.